lib/shipit/command.rb

```diff
@@ -183,18 +182,18 @@ def kill_and_wait(sig, wait, &block)
       rescue TimedOut
       rescue Errno::EIO # EIO is somewhat expected on Linux: http://stackoverflow.com/a/10306782
         # If we try to read the stream right after sending a signal, we often get an Errno::EIO.
-        if status = Process.wait(@pid, Process::WNOHANG)
-          return status
-        else
-          # If we let the child a little bit of time, it solves it.
-          retry_count -= 1
-          if retry_count > 0
-            sleep 0.05
-            retry
-          end
+        if reap_child!(block: false)
```

Comment:
This had no tests previously and is very difficult to test for - trying to reap a process not yet ready to exit. I've eyeballed this carefully and expect that I've preserved the existing behaviour, please check my working on this.